syntax = "proto3";
package woops.rpc;

service PsService {
    rpc CheckAlive(CheckAliveRequest) returns (CheckAliveResponse) {}
    rpc BarrierNotify(BarrierNotifyRequest) returns (BarrierNotifyResponse) {}
    rpc ForceSync(ForceSyncRequest) returns (ForceSyncResponse) {}
    rpc Update(stream UpdateRequest) returns (stream UpdateResponse) {}
    rpc Pull(stream PullRequest) returns (stream PullResponse) {}
    rpc Push(stream PushRequest) returns (stream PushResponse) {}
}

message CheckAliveRequest {}

message CheckAliveResponse {
    bool status = 1; 
}

message BarrierNotifyRequest {}

message BarrierNotifyResponse {}

message ForceSyncRequest {
    string tablename = 1;
    bytes parameter = 2;
}

message ForceSyncResponse {
}

message UpdateRequest {
    string tablename = 1;
    int32 iteration = 2;
    bytes delta = 3;
}

message UpdateResponse {
}

message PullRequest {
    string tablename = 1;
    int32 iteration = 2;
}

message PullResponse {}

message PushRequest {
    string tablename = 1;
    int32 iteration = 2;
    bytes parameter = 3;
}

message PushResponse {}

